### ISSUE TITLE
Support a `caption` with the Tables UI compnoent

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#11493]
### Is your proposal related to a problem?

While reviewing https://github.com/wagtail/wagtail/pull/11250 I found that the current `ui.tables.Table` component has no simple way to add a table `caption`. We kept the existing behaviour of using aria-description but a caption would be more suitable.

### Describe the solution you'd like

* `wagtail/admin/ui/tables/__init__.py` should have a default property of `caption` as an empty string.
* There should be a new method `get_caption` that returns the property, but this can be overridden if needed.
* This `caption` value should be available on the table instance passed into context.
* The base table template `wagtail/admin/templates/wagtailadmin/tables/table.html` should render the caption if it's truthy with the class `w-sr-only` by default.
* Maybe it's worth having a way to override this class at the `get_caption` or `@property` method and a way to override the class, this may make sense as a standalone template partial. `wagtail/admin/templates/wagtailadmin/tables/caption.html`
* Unit tests will be a must - they can be added to `wagtail/admin/tests/ui/test_tables.py`
* By default we should add some styling, even if it's for screen readers only, to make the caption show at the bottom https://developer.mozilla.org/en-US/docs/Web/CSS/caption-side -> this can probably be added to client/scss/generic/_normalize.scss
* You can validate the global styling in the styleguide also as we have a table caption here wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html

Once the above is done, either on the same PR or a different PR, replace the use of `aria-description` with the new `caption` for when the page listing is in sortable mode.

* See wagtail/admin/views/pages/listing.py
* ...`"Focus on the drag button`
* It should only set the caption if `show_ordering_column` is True.

### Describe alternatives you've considered

* Do nothing or keep using the aria-description even though this is not the most appropriate way to inform the user in screen readers.

### Additional context

* For work done in the ordering of page listing see #11250
* https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

* Would be good to have a basic understanding of the HTML caption element and Django templates + Django unit tests.

**Comment by salty-ivy** [comment]
I can pick it up?

**Comment by lb-** [comment]
All yours @salty-ivy if you are keen. Feel free to do in two parts if that's easier (first, add support for caption, second use the caption in the page listing ordering mode).

Or one PR will also be fine but try to split to two commits at least.

**Comment by salty-ivy** [comment]
@lb-  I think we can put the ordrable controller adaption in this PR only as it is a single line change in `wagtail/admin/views/pages/listing.py`

`kwargs["caption"] = "Focus on the drag button..."`

**Comment by lb-** [comment]
Yep. I think you're right. One PR is fine.
